### PR TITLE
Pact

### PR DESCRIPTION
**Pull request by jmsampayo**
Fixes #
contract.md in order to not contradict the "design-first" rule

## Proposed Changes

  - Adding PACT mandatory for CORE APIs and recommended for the rest of them

### CODE REVIEW

File: rest-api-guidelines/core-principles/testing.md [hunk 1]
```diff
@@ -1,6 +1,18 @@
 # Testing
 
+## DREED
```
Comment (kibertoad) [inline rest-api-guidelines/core-principles/testing.md]: Typo? "Dredd"?

File: rest-api-guidelines/functionality/message/error-reporting.md [hunk 2]
```diff
@@ -122,5 +124,8 @@ A Problem Detail response **MUST NOT** contain a program stack trace or server l
 
 ## Working with Problem Detail
 
-There are a whole plethora of libraries working with Problem Detail, for example, see [Zalando / Problem](https://github.com/zalando/problem) \(Java\).
+An API description **MAY** list all the error codes with which the API responds. The error responses **SHOULD** describre the error object model schema. It is **RECOMMENDED** to include examples of a possible error response. The error description and/or error example **MAY** list all the types of errors returned for a given error code.
```
Comment (kibertoad) [inline rest-api-guidelines/functionality/message/error-reporting.md]: "describre"

File: rest-api-guidelines/core-principles/testing.md (hunk 1 above)
Comment (jmsampayo) [reply rest-api-guidelines/core-principles/testing.md]: Ups. Sorry. Thanks! Corrected in new pull request.

**Comment by jmsampayo** [comment]
NodeJs and Java seeds in progress. We are co-creating them with several teams. We will add links to the seeds and examples asap.

**Pull request closed without merging**